### PR TITLE
Storage base path

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [ ] In Progress
- [x] Ready (see the comment about testing)
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40229 
fixes: https://github.com/demisto/etc/issues/37575

## Description
Removed the STORAGE_BASE_PATH constants from marketplace_constnats.py for the following reasons:
1. Avoid bad practice of changing constants
2. Allow using a `storage_base_path` argument as an argument for the `trigger test upload script` for allowing uploading to a specific path in the target bucket, and as part of that allow multiple uploads to the same bucket (which is not supported now).

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

**Comment by Noy-Maimon** [comment]
## Testing
Important notes about the following attached tests:
1. The testing via the gitlab flow will be added in once the issue with triggering the flow will be fixed.
2. The current version is using the OLD trigger test upload file (./Utils/trigger_test_upload_flow.sh) for testing, and once the new one will be merged (the one in this PR #13873), it will be modified too and tested again for verification.
3. In all of these tests the modified pack is `AbnormalSecurity` and the change is an added comment in the integration file + modifying an RN.

**Test cases and their green builds:**
1. Test of a modified pack uploaded to the test bucket (the usual case) - [circle build](https://app.circleci.com/pipelines/github/demisto/content/117956/workflows/bd8b8e63-02d5-4c43-8936-4050374f9300). 
2. Test of a modified pack uploaded to the test bucket to a specific storage base pack - to be added
3. Test of a modified pack uploaded to the production bucket - to be added
4. Test of force upload - to be added
5. Test of the private build - to be added

File: .gitlab/ci/bucket-upload.yml [hunk 1]
```diff
@@ -198,10 +208,18 @@ force-pack-upload:
     - CI_COMMIT_BRANCH=${CI_COMMIT_BRANCH:-unknown}
     - GCS_BUILD_BUCKET="marketplace-ci-build"
     - |
-      if [[ $GCS_MARKET_BUCKET != $GCS_PRODUCTION_BUCKET ]]; then
-          STORAGE_BASE_PATH="upload-flow/builds/$CI_COMMIT_BRANCH/$CI_PIPELINE_ID/content/packs"
+      if [[ $GCS_MARKET_BUCKET == $GCS_PRODUCTION_BUCKET ]] && [[ -z $STORAGE_BASE_PATH ]]; then
+        echo "DELETE AFTER TESTING: case 1: upload to prod bucket"
```
Comment (yaakovi) [inline .gitlab/ci/bucket-upload.yml:212]: TODO

File: .gitlab/ci/bucket-upload.yml [hunk 2]
```diff
@@ -198,10 +208,18 @@ force-pack-upload:
...
+        STORAGE_BASE_PATH="content"
+      fi
+
+      if [[ $GCS_MARKET_BUCKET != $GCS_PRODUCTION_BUCKET ]] && [[ -z $STORAGE_BASE_PATH ]]; then
+       STORAGE_BASE_PATH="upload-flow/builds/$CI_COMMIT_BRANCH/$CI_PIPELINE_ID/content"
+       echo "DELETE AFTER TESTING: case 2: create a new target bucket"
       fi
-    - python3 ./Tests/Marketplace/copy_and_upload_packs.py -a $PACK_ARTIFACTS -e $EXTRACT_FOLDER -pb "$GCS_MARKET_BUCKET" -bb "$GCS_BUILD_BUCKET" -s $GCS_MARKET_KEY -n $CI_PIPELINE_ID -c $CI_COMMIT_BRANCH -p "${PACKS_TO_UPLOAD}" -pbp "${STORAGE_BASE_PATH}"
+       echo "DELETE AFTER TESTING: if non of the other cases was printed, we are in case 3: the default, the inserted value to storage_base_path is the one being transfered to the copy_and_upload script"
```
Comment (yaakovi) [inline .gitlab/ci/bucket-upload.yml:220]: TODO

**Review by yaakovi — COMMENTED** [review]
I marked all the testing code so we can track and revert it before merging

File: Tests/scripts/prepare_content_packs_for_testing.sh [hunk 3]
```diff
@@ -32,19 +32,25 @@ else
...
-  # ====== UPDATING TESTING BUCKET ======
+  if [[ -z "$2" ]]; then  # case 2: a test upload, the source is of a new target bucket, need to copy from prod to that target
+    SOURCE_PATH="upload-flow/builds/$CI_COMMIT_BRANCH/$CI_PIPELINE_ID/content"
+    # ====== UPDATING TESTING BUCKET ======
+    echo "Copying production bucket files at: gs://$GCS_PRODUCTION_BUCKET/content to testing bucket at path: gs://$GCS_MARKET_BUCKET/$SOURCE_PATH ..."
+    gsutil -m cp -r "gs://$GCS_PRODUCTION_BUCKET/content" "gs://$GCS_MARKET_BUCKET/$SOURCE_PATH" > "$ARTIFACTS_FOLDER/logs/Prepare Content Packs For Testing gsutil.log" 2>&1
+    echo "Finished copying successfully."
+    # ====== UPDATING TESTING BUCKET ======
+  else  # case 3: a test upload, the source is of an exiting target bucket, no need to copy from the prod bucket TODO: add documentation
+    SOURCE_PATH=$2  # should look like: "upload-flow/builds/$CI_COMMIT_BRANCH/$CI_PIPELINE_ID/content"
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply Tests/scripts/prepare_content_packs_for_testing.sh:46]: I was thinking to add the validation on the triggering script, as in the one that will be added on itay's PR (https://github.com/demisto/content/pull/13873). Do you think here is prefered?

File: Tests/scripts/validate_premium_packs.sh [hunk 4]
```diff
@@ -10,22 +10,19 @@ SECRET_CONF_PATH="./conf_secret.json"
 
 # ====== BUCKET CONFIGURATION ======
 
-if [[ $GCS_MARKET_BUCKET != $GCS_PRODUCTION_BUCKET ]]; then
-  STORAGE_BASE_PATH="upload-flow/builds/$BRANCH/$CI_PIPELINE_ID/content/packs"
+if [[ -z $STORAGE_BASE_PATH ]]; then
+  if [[ $GCS_MARKET_BUCKET == $GCS_PRODUCTION_BUCKET ]]; then
+   STORAGE_BASE_PATH="content"
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply Tests/scripts/validate_premium_packs.sh:15]: The path without the `/packs` extra is being used as is on `prepare_content_packs_for_testing.sh`. So either I trim the `packs` from the path on that script, or I add it when using these scripts. For me it was clearer this way, what to you think?

File: Tests/Marketplace/marketplace_constants.py [hunk 5]
```diff
@@ -39,7 +39,9 @@ class GCPConfig(object):
     """ Google cloud storage basic configurations
 
     """
-    STORAGE_BASE_PATH = "content/packs"  # configurable base path for packs in gcs, can be modified
+    CONTENT_PACKS_PATH = "content/packs"
+    # STORAGE_BASE_PATH = "content/packs"  TODO: remove
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply Tests/Marketplace/marketplace_constants.py]: No actual changes in the private build other than removing the use of this constant. The constant was replaced in each case with the suitable replacement. This should be tested but from my understanding there is no way to test trigger it, so will have to verify after merging - correct me if I'm wrong.

File: Tests/scripts/validate_premium_packs.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (guyfreund) [reply Tests/scripts/validate_premium_packs.sh:15]: Whatever you guys decide, make sure that the copying from target to build copies from `content` included and not just from `packs` onwards.

File: Utils/trigger_test_upload_flow.sh [hunk 6]
```diff
@@ -11,6 +11,8 @@ if [ "$#" -lt "1" ]; then
   [-p, --packs]               CSV list of pack IDs. Mandatory when the --force flag is on.
   [-ch, --slack-channel]      A slack channel to send notifications to. Default is dmst-bucket-upload.
   [-g, --gitlab]              Flag indicating to trigger the flow in GitLab.
+  [-sbp, --storage-base-path] a path to copy from and upload to in this current upload.
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply Utils/trigger_test_upload_flow.sh:14]: It should me mentioned that this path is also the target path, as in the packs will be uploaded to that path as the target bucket.
maybe:
```suggestion
  [-sbp, --storage-base-path]  a path to copy from in this current upload, and to be used as a target destination.
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -1243,27 +1244,25 @@ jobs:
...
                 name: Download packs from GCP
                 command: |
-                  if [[ $GCS_MARKET_BUCKET == $GCS_PRODUCTION_BUCKET ]] && [[ -z $STORAGE_BASE_PATH ]]; then
-                     STORAGE_BASE_PATH="content/packs"
-                  fi
-                  if [[ $GCS_MARKET_BUCKET != $GCS_PRODUCTION_BUCKET ]] && [[ -z $STORAGE_BASE_PATH ]]; then
-                    STORAGE_BASE_PATH="upload-flow/builds/$CI_COMMIT_BRANCH/$CI_PIPELINE_ID/content/packs"
+                  if [[ -z $STORAGE_BASE_PATH ]]; then
+                    if [[ $GCS_MARKET_BUCKET == $GCS_PRODUCTION_BUCKET ]]; then
+                     STORAGE_BASE_PATH="content"
```
Comment (yaakovi) [inline .circleci/config.yml:1259]: indentation fix
```suggestion
                      STORAGE_BASE_PATH="content"
```

**Review by yaakovi — APPROVED** [review]
nice work

File: Tests/scripts/prepare_content_packs_for_testing.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply Tests/scripts/prepare_content_packs_for_testing.sh:46]: added validations for the path in `trigger_test_upload_flow.sh`

**Pull request merged**